<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogNPCEdit</class>
 <widget class="QDialog" name="DialogNPCEdit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1018</width>
    <height>698</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="0" column="0">
    <widget class="QWidget" name="gridWidget" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QWidget" name="horizontalWidget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="sizeConstraint">
          <enum>QLayout::SetNoConstraint</enum>
         </property>
         <property name="rightMargin">
          <number>20</number>
         </property>
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <property name="bottomMargin">
            <number>20</number>
           </property>
           <item row="0" column="1" colspan="2">
            <widget class="QLineEdit" name="NpcName">
             <property name="text">
              <string>My NPC</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Graphic Width: </string>
             </property>
            </widget>
           </item>
           <item row="4" column="2">
            <widget class="QSpinBox" name="npcGraphicSizeSpin_h">
             <property name="minimum">
              <number>8</number>
             </property>
             <property name="maximum">
              <number>128</number>
             </property>
             <property name="value">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QSpinBox" name="NpcHP">
             <property name="value">
              <number>5</number>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Graphic Height:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QComboBox" name="npcListCombobox"/>
           </item>
           <item row="3" column="2">
            <widget class="QSpinBox" name="npcGraphicSizeSpin_w">
             <property name="minimum">
              <number>8</number>
             </property>
             <property name="maximum">
              <number>128</number>
             </property>
             <property name="value">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Name:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Graphic:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Hit Points:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QCheckBox" name="checkBoxCanShoot">
             <property name="text">
              <string>Can Shoot</string>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>*IA Type:</string>
             </property>
            </widget>
           </item>
           <item row="7" column="2">
            <widget class="QComboBox" name="comboBoxIA"/>
           </item>
           <item row="12" column="2">
            <widget class="QDialogButtonBox" name="buttonBox">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="standardButtons">
              <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
             </property>
            </widget>
           </item>
           <item row="6" column="2">
            <widget class="QComboBox" name="comboBoxShieldType">
             <item>
              <property name="text">
               <string>no shield</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>full shield</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>front shield</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string/>
              </property>
             </item>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Shield:</string>
             </property>
            </widget>
           </item>
           <item row="11" column="0">
            <widget class="QCheckBox" name="isBossCheckbox">
             <property name="text">
              <string>Is Boss</string>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Speed:</string>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="9" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Range:</string>
             </property>
            </widget>
           </item>
           <item row="8" column="2">
            <widget class="QSpinBox" name="speed_spin">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="value">
              <number>2</number>
             </property>
            </widget>
           </item>
           <item row="9" column="2">
            <widget class="QSpinBox" name="range_spin">
             <property name="minimum">
              <number>48</number>
             </property>
             <property name="maximum">
              <number>320</number>
             </property>
             <property name="value">
              <number>106</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QWidget" name="verticalWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="styleSheet">
            <string notr="true">border: 1px solid green;</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <property name="leftMargin">
             <number>10</number>
            </property>
            <property name="topMargin">
             <number>1</number>
            </property>
            <item>
             <widget class="QWidget" name="verticalWidget_2" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="styleSheet">
               <string notr="true">border: 2px solid black;</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <property name="margin">
                <number>0</number>
               </property>
               <item>
                <widget class="QScrollArea" name="scrollArea">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="verticalScrollBarPolicy">
                  <enum>Qt::ScrollBarAlwaysOn</enum>
                 </property>
                 <property name="horizontalScrollBarPolicy">
                  <enum>Qt::ScrollBarAlwaysOn</enum>
                 </property>
                 <property name="widgetResizable">
                  <bool>false</bool>
                 </property>
                 <widget class="QWidget" name="scrollAreaWidgetContents">
                  <property name="geometry">
                   <rect>
                    <x>0</x>
                    <y>0</y>
                    <width>434</width>
                    <height>558</height>
                   </rect>
                  </property>
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                    <horstretch>100</horstretch>
                    <verstretch>100</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="autoFillBackground">
                   <bool>false</bool>
                  </property>
                  <widget class="npcPreviewArea" name="npcPreviewAreaWidget" native="true">
                   <property name="geometry">
                    <rect>
                     <x>0</x>
                     <y>0</y>
                     <width>651</width>
                     <height>581</height>
                    </rect>
                   </property>
                   <property name="styleSheet">
                    <string notr="true">border: 2px solid black;</string>
                   </property>
                  </widget>
                 </widget>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="label_4">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Duration:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox">
                <property name="minimum">
                 <number>5</number>
                </property>
                <property name="maximum">
                 <number>9999</number>
                </property>
                <property name="value">
                 <number>20</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QComboBox" name="frameTypeCombo">
              <item>
               <property name="text">
                <string>NONE</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_STAND</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_WALK</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_JUMP</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_ATTACK</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_TELEPORT</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_JUMP_ATTACK</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_WALK_ATTACK</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_HIT</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_STAIRS</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_STAIRS_SEMI</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_STAIRS_ATTACK</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_STAIRS_MOVE</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_TROW</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_TURN</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_MOVE_SEMI</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_WALK_AIR</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_INTRO</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>ANIM_SPECIAL_ATTACK</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>npcPreviewArea</class>
   <extends>QWidget</extends>
   <header>npcpreviewarea.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogNPCEdit</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogNPCEdit</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>finishedNPCEditor()</signal>
 </slots>
</ui>
